subsys/shell/shell_uart.c

```diff
@@ -85,10 +80,7 @@ static void uart_rx_handle(struct device *dev,
 			/* If successful in getting byte from the fifo, try
 			 * feeding it to SMP as a part of mcumgr frame.
 			 */
-			if ((rd_len != 0) &&
-			    smp_shell_rx_byte(&sh_uart->ctrl_blk->smp, dummy)) {
-				new_data = true;
-			}
+			smp_shell_rx_bytes(&sh_uart->ctrl_blk->smp, &dummy, 1);
```

Comment:
This seems to revert your recent PR. I don't think this was intentional.

Reply:
Yes that is botched rebase.

Reply:
Fixed.